testutil/keeper/proof.go

```diff
@@ -45,50 +58,129 @@ func ProofKeeper(
+	keys := storetypes.NewKVStoreKeys(
+		types.StoreKey,
+		sessiontypes.StoreKey,
+		suppliertypes.StoreKey,
+		apptypes.StoreKey,
+		gatewaytypes.StoreKey,
+	)
+
+	// Construct a multistore & mount store keys for each keeper that will interact with the state store.
+	stateStore := integration.CreateMultiStore(keys, log.NewNopLogger())
+
+	logger := log.NewTestLogger(t)
+	ctx := sdk.NewContext(stateStore, cmtproto.Header{}, false, logger)
+
+	// Set block height to 1 so there is a valid session on-chain.
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	ctx = sdkCtx.WithBlockHeight(1)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+	authority := authtypes.NewModuleAddress(govtypes.ModuleName)
+
+	// Mock the bank keeper.
+	ctrl := gomock.NewController(t)
+	bankKeeperMock := mocks.NewMockBankKeeper(ctrl)
+
+	// Construct a real account keeper so that public keys can be queried.
+	accountKeeper := authkeeper.NewAccountKeeper(
+		cdc,
+		runtime.NewKVStoreService(keys[authtypes.StoreKey]),
+		authtypes.ProtoBaseAccount,
+		map[string][]string{minttypes.ModuleName: {authtypes.Minter}},
+		addresscodec.NewBech32Codec(app.AccountAddressPrefix),
+		app.AccountAddressPrefix,
+		authority.String(),
+	)
+
+	// Construct gateway keeper with a mocked bank keeper.
+	gatewayKeeper := gatewaykeeper.NewKeeper(
+		cdc,
+		runtime.NewKVStoreService(keys[gatewaytypes.StoreKey]),
+		logger,
+		authority.String(),
+		bankKeeperMock,
+	)
+	require.NoError(t, gatewayKeeper.SetParams(ctx, gatewaytypes.DefaultParams()))
+
+	// Construct an application keeper to add apps to sessions.
+	appKeeper := appkeeper.NewKeeper(
+		cdc,
+		runtime.NewKVStoreService(keys[apptypes.StoreKey]),
+		logger,
+		authority.String(),
+		bankKeeperMock,
+		accountKeeper,
+		gatewayKeeper,
+	)
+	require.NoError(t, appKeeper.SetParams(ctx, apptypes.DefaultParams()))
+
+	// Construct a real supplier keeper to add suppliers to sessions.
+	supplierKeeper := supplierkeeper.NewKeeper(
+		cdc,
+		runtime.NewKVStoreService(keys[suppliertypes.StoreKey]),
+		log.NewNopLogger(),
+		authority.String(),
+		bankKeeperMock,
+	)
+	require.NoError(t, supplierKeeper.SetParams(ctx, suppliertypes.DefaultParams()))
+
+	// Construct a real session keeper so that sessions can be queried.
+	sessionKeeper := sessionkeeper.NewKeeper(
+		cdc,
+		runtime.NewKVStoreService(keys[sessiontypes.StoreKey]),
+		log.NewNopLogger(),
+		authority.String(),
+		accountKeeper,
+		bankKeeperMock,
+		appKeeper,
+		supplierKeeper,
+	)
+	require.NoError(t, sessionKeeper.SetParams(ctx, sessiontypes.DefaultParams()))
+
+	proofKeeper, err := keeper.NewKeeper(
+		cdc,
+		runtime.NewKVStoreService(keys[types.StoreKey]),
+		log.NewNopLogger(),
+		authority.String(),
+		sessionKeeper,
+		appKeeper,
+		accountKeeper,
+	)
+	require.NoError(t, err)
+	require.NoError(t, proofKeeper.SetParams(ctx, types.DefaultParams()))
+
+	keeperWithDeps := ProofKeeperWithDeps{
```

Comment:
IMO we should remove the `ProofKeeperWithDeps` struct and return a a list of all the keepers.

Reasoning:
1. We do/could to decompose it on the caller side anyhow
2. Aside from being an envelope, it doesn't seem to provide any benefit on the abstraction side
3. It's a little misleading what `Deps` is referring to

Reply:
See https://github.com/pokt-network/poktroll/pull/412#discussion_r1519259812